lib/rig/event_filter.ex

```diff
@@ -291,9 +291,9 @@ defmodule Rig.EventFilter do
 
   # ---
 
-  @callback forward_event(CloudEvent.t()) :: :ok
-  @spec forward_event(CloudEvent.t()) :: :ok
-  def forward_event(%CloudEvent{} = event) do
+  @callback forward_event(Cloudevents.t()) :: :ok
+  @spec forward_event(Cloudevents.t()) :: :ok
+  def forward_event(%{} = event) do
```

Comment:
This is rather un-idiomatic - this is usually done with an `when is_map(event)` guard. However, I'm not sure this is really helpful either way. Haven't tried to write a guard myself so far - is it possible to do something like this:

```elixir
def forward_event(event) when Cloudevents.is_cloudevent(event) do
```

?

Reply:
Got

```bash
== Compilation error in file lib/rig/event_filter.ex ==
** (CompileError) lib/rig/event_filter.ex:296: cannot invoke remote function Cloudevents.is_cloudevent/1 inside guards
```